package.json

```diff
@@ -1,17 +1,15 @@
 {
   "name": "rancher-extension-stackstate",
-  "version": "0.3.0",
   "license": "Apache-2.0",
-  "private": false,
+  "private": true,
```

Comment:
I belive the top-level `package.json` does not have anything to do with public packages inside `pkg/` folder, so it can be private and loose version field

Reply:
I don't know if this has any impact with the sync mechanism that allows us to publish the extension on the official repo, but we can give it a try. Worse case scenario, we have to revert this

Reply:
IIUC the top-level package version is used when [publishing the Extension Catalog Image](https://extensions.rancher.io/extensions/publishing): _"An ECI contains the Helm charts and assets within the image, and is determined by the main Extension version property found in ./package.json"_

Reply:
That's interesting...

What does happen if a top-level version is different from `pkg/*/package.json` version(s)? There is no guarantee that they are always aligned

Reply:
this repo only has `workfows/build-extension-charts.yml`, so seems like `./package.json` version is not in use 🤔

Reply:
IIUC the top level version is basically a version of the ECI - of the Helm chart repository it provides (ECI is used to create a local Helm chart repository for disconnected installations) The actual extension package versions it includes can be different. @jordojordo please correct me if I am wrong.

Reply:
@jtomasek is correct. The top level package version is only used when creating an ECI, which would occur in the build-extension-catalog workflow. I've [removed this workflow](https://github.com/StackVista/rancher-extension-stackstate/commit/3ac2bdc5d171112110b88f8c0dba95f43eb02e7f) previously as it isn't necessary within this repository.

Reply:
I would say that the best practice is to keep them the same and always updated in both sides... Do you agree @jordojordo ?

Reply:
@owanturist That looks like the correct flow, but ultimately it's up to you. If you plan on creating more extension packages then what you suggest would be the best way to keep things succinct.

Whenever you add the extension to our [ui-plugin-charts manifest](https://github.com/rancher/ui-plugin-charts/blob/main/manifest.json) you will need to create an entry for the extension package name, and the subsequent versions that are related to that package.

Reply:
@aalves08 That would be best practice at least until there are multiple extension packages. But for the time being that would be the best approach.

Reply:
all clear, appreciate the guidance 🙏